.github/workflows/linting.yml

```diff
@@ -0,0 +1,47 @@
+name: Lint scripts & Markdown
+
+on: [push, pull_request, workflow_call]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  run_linters:
+    name: Run scripts & Markdown linters
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          submodules: false
+
+      - name: Update packages
+        run:  sudo apt-get update
+
+      - name: Run shellcheck
+        run:  ./scripts/verify-bash.sh
+
+      - name: Install pylint
+        run: |
+          sudo apt-get install python3-setuptools
+          sudo pip3 install pylint beautifulsoup4 html5lib
+
+      - name: Run pylint
+        run:  ./scripts/verify-python.sh
+
+      - name: Install markdownlint
+        run: |
+          sudo apt-get install ruby-full
+          ruby --version
+          sudo gem install mdl
+
+      - name: Run markdownlint
+        run:  ./scripts/verify-markdown.sh
+
+      - name: Install appstream-util
+        run:  sudo apt-get install appstream-util
+
+      - name: Verify metainfo.xml
+        run:  appstream-util validate-relax --nonet contrib/linux/dosbox-staging.metainfo.xml
```

Comment:
Since you are moving this out to it's own workflow anyway, how about simplifying the workflow to install all required packages in a single step?

It's fine if you don't want to do that in this PR, but it was something I just noticed, so I'm mentioning it here.

Reply:
Although the current setup does allow for "early abort" I suppose.

Reply:
I slightly prefer the current approach because these steps also install Python and Ruby-specific stuff with `pip` and `gem`, respectively. If something fails, it's at least a bit easier to see where exactly the problem occurred.

Also, I don't think merging this all into a single step would add any meaningful speed benefits (this runs very quickly currently anyway). Leaving as is for now.